File: ios/WorkApp/Views/Components.swift
import SwiftUI

/// Initials on a department-coloured circle, with ink picked for contrast.
struct AvatarView: View {
    let name: String
    var size: CGFloat = 40
    var hex: String = "#8E8E93"

    private var initials: String {
        name.split(separator: " ").prefix(2).compactMap { $0.first }.map(String.init).joined()
    }

    var body: some View {
        Circle()
            .fill(Color(hex: hex))
            .frame(width: size, height: size)
            .overlay(
                Text(initials.isEmpty ? "?" : initials)
                    .font(.system(size: size * 0.38, weight: .bold))
                    .foregroundStyle(Color.ink(on: hex)))
    }
}

/// A small status pill — call status, counts, labels.
struct Pill: View {
    let text: String
    var hex: String?

    var body: some View {
        Text(text)
            .font(.caption.weight(.semibold))
            .padding(.horizontal, 9)
            .padding(.vertical, 3)
            .background(hex.map { Color(hex: $0) } ?? Color(.tertiarySystemFill),
                        in: Capsule())
            .foregroundStyle(hex.map { Color.ink(on: $0) } ?? Color.secondary)
    }
}

/// The quick-dial chip used in the contact bar on the Today screen.
struct ContactChip: View {
    let entry: RosterEntry
    var onTap: () -> Void

    var body: some View {
        Button(action: onTap) {
            VStack(spacing: 6) {
                AvatarView(name: entry.person.name, size: 54, hex: entry.person.dept.hex)
                    .overlay(alignment: .topTrailing) {
                        Circle()
                            .fill(Color(hex: entry.call.status.hex))
                            .frame(width: 14, height: 14)
                            .overlay(Circle().stroke(Color(.systemBackground), lineWidth: 2.5))
                            .offset(x: 2, y: -2)
                    }
                Text(entry.person.name)
                    .font(.system(size: 11, weight: .semibold))
                    .lineLimit(2)
                    .multilineTextAlignment(.center)
                if let role = entry.roleLabels.first {
                    Text(role)
                        .font(.system(size: 10))
                        .foregroundStyle(.secondary)
                        .lineLimit(1)
                }
            }
            .frame(width: 68)
        }
        .buttonStyle(.plain)
        .accessibilityLabel("\(entry.person.name), \(entry.roleLabels.joined(separator: " "))")
    }
}

/// A tappable row for a spreadsheet column that may or may not be filled.
struct SlotRow: View {
    let label: String
    let person: Person?
    let time: String?
    var onTap: () -> Void
    var onCall: (() -> Void)?

    var body: some View {
        HStack(spacing: 12) {
            if let person {
                AvatarView(name: person.name, size: 38, hex: person.dept.hex)
            } else {
                Circle()
                    .fill(Color(.tertiarySystemFill))
                    .frame(width: 38, height: 38)
                    .overlay(Image(systemName: "plus").foregroundStyle(.tertiary))
            }

            Button(action: onTap) {
                VStack(alignment: .leading, spacing: 1) {
                    Text(person?.name ?? "לא שובץ")
                        .font(.body)
                        .foregroundStyle(person == nil ? .tertiary : .primary)
                    Text(label)
                        .font(.footnote)
                        .foregroundStyle(.secondary)
                }
                .frame(maxWidth: .infinity, alignment: .leading)
                .contentShape(Rectangle())
            }
            .buttonStyle(.plain)

            if let time, !time.isEmpty {
                Text(time)
                    .font(.subheadline.monospacedDigit())
                    .foregroundStyle(.secondary)
                    .environment(\.layoutDirection, .leftToRight)
            }

            if let onCall {
                Button(action: onCall) {
                    Image(systemName: "phone.fill")
                }
                .buttonStyle(.plain)
                .foregroundStyle(Color.accentColor)
                .accessibilityLabel("חיוג ל\(person?.name ?? "")")
            }
        }
    }
}

/// Empty-state block used across the tabs.
struct EmptyState: View {
    let icon: String
    let title: String
    let message: String
    var actionTitle: String?
    var action: (() -> Void)?

    var body: some View {
        VStack(spacing: 12) {
            Image(systemName: icon)
                .font(.system(size: 42))
                .foregroundStyle(.tertiary)
            Text(title).font(.title3.bold())
            Text(message)
                .font(.subheadline)
                .foregroundStyle(.secondary)
                .multilineTextAlignment(.center)
            if let actionTitle, let action {
                Button(actionTitle, action: action)
                    .buttonStyle(.borderedProminent)
                    .padding(.top, 4)
            }
        }
        .frame(maxWidth: .infinity)
        .padding(.horizontal, 32)
        .padding(.vertical, 44)
    }
}

/// Digits stay in dialling order inside an otherwise right-to-left layout.
struct LTRText: View {
    let text: String
    var font: Font = .subheadline

    var body: some View {
        Text(text)
            .font(font)
            .environment(\.layoutDirection, .leftToRight)
    }
}

File: android/app/src/main/java/com/workapp/hafaka/ui/Components.kt
package com.workapp.hafaka.ui

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.LocalLayoutDirection
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.LayoutDirection
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.compose.runtime.CompositionLocalProvider
import com.workapp.hafaka.model.RosterEntry

/** Initials on a department-coloured circle, with ink picked for contrast. */
@Composable
fun Avatar(name: String, size: Int = 40, hex: String = "#8E8E93") {
    val initials = name.trim().split(" ").filter { it.isNotBlank() }
        .take(2).mapNotNull { it.firstOrNull() }.joinToString("")
    Box(
        Modifier.size(size.dp).clip(CircleShape).background(colorOf(hex)),
        contentAlignment = Alignment.Center,
    ) {
        Text(
            initials.ifBlank { "?" },
            color = inkOn(hex),
            fontSize = (size * 0.38).sp,
            fontWeight = FontWeight.Bold,
        )
    }
}

/** A small status pill — call status, counts, labels. */
@Composable
fun Pill(text: String, hex: String? = null) {
    Text(
        text,
        fontSize = 12.sp,
        fontWeight = FontWeight.SemiBold,
        color = hex?.let { inkOn(it) } ?: MaterialTheme.colorScheme.onSurfaceVariant,
        modifier = Modifier
            .clip(CircleShape)
            .background(hex?.let { colorOf(it) } ?: MaterialTheme.colorScheme.surfaceVariant)
            .padding(horizontal = 9.dp, vertical = 3.dp),
    )
}

/** Digits keep dialling order inside an otherwise right-to-left layout. */
@Composable
fun LtrText(text: String, style: androidx.compose.ui.text.TextStyle = MaterialTheme.typography.bodySmall,
            color: Color = MaterialTheme.colorScheme.onSurfaceVariant) {
    CompositionLocalProvider(LocalLayoutDirection provides LayoutDirection.Ltr) {
        Text(text, style = style, color = color)
    }
}

/** The grouped card that every list sits in. */
@Composable
fun Card(modifier: Modifier = Modifier, content: @Composable ColumnScope.() -> Unit) {
    Column(
        modifier
            .fillMaxWidth()
            .padding(horizontal = 16.dp)
            .clip(RoundedCornerShape(14.dp))
            .background(MaterialTheme.colorScheme.surface),
        content = content,
    )
}

@Composable
fun SectionTitle(text: String, trailing: String? = null) {
    Row(
        Modifier.fillMaxWidth().padding(start = 20.dp, end = 20.dp, top = 18.dp, bottom = 7.dp),
        horizontalArrangement = Arrangement.SpaceBetween,
        verticalAlignment = Alignment.CenterVertically,
    ) {
        Text(text, style = MaterialTheme.typography.labelLarge,
             color = MaterialTheme.colorScheme.onSurfaceVariant)
        trailing?.let {
            Text(it, style = MaterialTheme.typography.labelLarge,
                 color = MaterialTheme.colorScheme.onSurfaceVariant)
        }
    }
}

@Composable
fun RowDivider() {
    HorizontalDivider(
        Modifier.padding(start = 68.dp),
        thickness = 0.5.dp,
        color = MaterialTheme.colorScheme.outlineVariant,
    )
}

/** The quick-dial chip in the contact bar on the Today screen. */
@Composable
fun ContactChip(entry: RosterEntry, onClick: () -> Unit) {
    Column(
        Modifier.width(68.dp).clickable(onClick = onClick),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(6.dp),
    ) {
        Box {
            Avatar(entry.person.name, 54, entry.person.dept.hex)
            Box(
                Modifier
                    .align(Alignment.TopEnd)
                    .size(14.dp)
                    .border(2.5.dp, MaterialTheme.colorScheme.background, CircleShape)
                    .padding(2.5.dp)
                    .clip(CircleShape)
                    .background(colorOf(entry.call.status.hex)),
            )
        }
        Text(
            entry.person.name,
            fontSize = 11.sp,
            fontWeight = FontWeight.SemiBold,
            textAlign = TextAlign.Center,
            maxLines = 2,
            overflow = TextOverflow.Ellipsis,
            color = MaterialTheme.colorScheme.onBackground,
        )
        entry.roleLabels.firstOrNull()?.let {
            Text(it, fontSize = 10.sp, maxLines = 1,
                 color = MaterialTheme.colorScheme.onSurfaceVariant)
        }
    }
}

@Composable
fun EmptyState(
    title: String,
    message: String,
    actionLabel: String? = null,
    onAction: (() -> Unit)? = null,
) {
    Column(
        Modifier.fillMaxWidth().padding(horizontal = 32.dp, vertical = 48.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(10.dp),
    ) {
        Text(title, style = MaterialTheme.typography.titleLarge)
        Text(message, style = MaterialTheme.typography.bodyMedium,
             color = MaterialTheme.colorScheme.onSurfaceVariant,
             textAlign = TextAlign.Center)
        if (actionLabel != null && onAction != null) {
            Button(onClick = onAction, modifier = Modifier.padding(top = 6.dp)) { Text(actionLabel) }
        }
    }
}
